probe-rs/src/debug/registers.rs

```diff
@@ -82,28 +102,52 @@ impl DebugRegisters {
 
         let mut debug_registers = Vec::<DebugRegister>::new();
 
-        let all_registers = [
-            (RegisterGroup::Base, register_file.platform_registers),
-            (RegisterGroup::Argument, register_file.argument_registers),
-            (RegisterGroup::Result, register_file.result_registers),
+        let mut all_registers = vec![
+            (
+                RegisterGroup::Base,
+                register_file.platform_registers.to_owned(),
+            ),
+            (
+                RegisterGroup::Argument,
+                register_file.argument_registers.to_owned(),
+            ),
+            (
+                RegisterGroup::Result,
+                register_file.result_registers.to_owned(),
+            ),
             (
                 RegisterGroup::Singleton,
-                &[register_file.frame_pointer.to_owned()],
+                [register_file.frame_pointer.to_owned()].to_vec(),
             ),
             (
                 RegisterGroup::Singleton,
-                &[register_file.program_counter.to_owned()],
+                [register_file.program_counter.to_owned()].to_vec(),
             ),
             (
                 RegisterGroup::Singleton,
-                &[register_file.return_address.to_owned()],
+                [register_file.return_address.to_owned()].to_vec(),
             ),
             (
                 RegisterGroup::Singleton,
-                &[register_file.stack_pointer.to_owned()],
+                [register_file.stack_pointer.to_owned()].to_vec(),
             ),
         ];
 
+        // Add additional registers required to unwind beyond the most recent signal handler.
+        if let Some(psr_register) = register_file.psr {
+            let mut psr = psr_register.to_owned();
+            all_registers.push((RegisterGroup::Singleton, [psr.clone()].to_vec()));
```

Comment:
I know that it wasn't added in this PR, but I don't understand what `Singleton` is supposed to mean in the context of registers.

For me, singleton means that there is only one of something, but that generally applies to all registers?